### ISSUE TITLE
Pester robs child powershell processes of stdout

### ISSUE THREAD

**Issue by johndog** [pester/Pester#1626]
## 1. General summary of the issue

Trying to run a powershell script in a child process, and test its outputs, including stderr, is stymied by the apparent fact that when running them under pester, stderr is somehow prevented from going anywhere.

## 2. Describe Your Environment
Windows 10 

```
Pester version     : 4.10.1 C:\Program Files\WindowsPowerShell\Modules\Pester\4.10.1\Pester.psd1
PowerShell version : 5.1.20161.1000
OS version         : Microsoft Windows NT 10.0.20161.0
```

## 3. Expected Behavior

If I run `cmd /c "powershell -c {somecommand} 2>&1"`, and the powershell invocation generates information for stderr, I should be able to capture that.

## 4. The bug

To repro, run this script from the command-line or with F5 in VSCode (beware that it will write two files in $PWD):

```
# Probably not germane, but don't leave it as a chance variable
$ErrorActionPreference = "Continue"

# Pester 5 runs code outside of Describe/Context twice, on
function AreWeRunningInPester
{
    @(Get-PSCallStack).Command -icontains "Invoke-Pester"
}

# Creates a .cmd script, runs it, confirms that the output file created with redirection in that script contains a string
# Warning:  overwrites two files:  $pwd\test1234.cmd and $pwd\out1234
function RunTestScenario
{
    $testCmd = "$pwd\test1234.cmd"
    $outFile = "$pwd\out1234"
   
    "" | set-content $pwd\out
    "$pwd\out" | should -Not -FileContentMatch "yikes"

    @"
powershell -c ". {write-error -ErrorAction Stop 'yikes'}" 2>&1 1>$outFile
"@ | set-content $testCmd

    cmd /c $testCmd
    "$outFile" | should -FileContentMatch "yikes"
}

# Run the test once "outside" of pester
if (!(AreWeRunningInPester))
{
    RunTestScenario
    "Success" | Out-Host
}

# Run the test again "inside" pester
Describe "same scenario" {
    It "does not work in pester" {
        RunTestScenario
    }
}
```

The part of the script _before_ the Describe block will run outside of pester.  It will succeed, showing that the cmd script did indeed capture stderr output from the powershell process.

With an F5 invocation, the part of the script _inside_ the Describe block will still run inside Pester.  The test will fail: the data captured by the cmd process from the powershell process is indeed empty.

If you run the whole script with Invoke-Pester, the first invocation of RunTestScenario will fail instead of succeeding.

I don't know how pester could be influencing child processes like this.

## 6. Context

I've been trying to compose tests around script and tool behaviors, and this bug is making it extremely difficult to test and/or use error handling in the process of those tests.  It's not just about write-error, if powershell generates a syntax error that seems to get lost, too.

**Comment by johndog** [comment]
I've determined that this is most likely an issue native to powershell.  Pester is not needed to reproduce it, all that's required is to capture RunTestScenario to a variable, a la:

```
$output = RunTestScenario
```

This alone will cause redirected outputs in child processes of RunTestScenario to be stolen from those children.

**Comment by nohwnd** [comment]
@johndog you are right, all output goes to $null in v4 and to $StandardOutput in v5. This is by design.

**Comment by johndog** [comment]
@nohwnd I assume you are talking about v4 and v5 of Pester.  As I wrote last, this reproduces outside of Pester, so I think you might be misreading the description.  

Bug paraphrased:   If Process1 launches Process2, which launches Process3, this bug is about Process1 somehow preventing Process2 from reliably redirecting the output of Process3.  In fact, I believe it only happens if Process3 itself is a powershell process; if it's anything else, redirection between Process2 and Process3 works correctly.  I understand that Process1, if it's pester, would redirect any output it gets to $null, but in my scenario Process2 has already redirected the output to a file so Pester wouldn't see it anyway.

**Comment by nohwnd** [comment]
@johndog you are right. I was misreading it. But let me rephrase as well 🙂 

In Pester v4 all output is assigned to $null and in Pester v5 it is assigned to $standardOutput (which is then assigned to StandardOutput property on the result object). This is by design and would cause the issue you describe above.
